sormas-ui/src/main/java/de/symeda/sormas/ui/immunization/components/form/ImmunizationDataForm.java

```diff
@@ -369,6 +369,11 @@ protected void onCancel() {
 			recoveryHeadingLabel.setVisible(isRecoveryVisible);
 			positiveTestResultDate.setVisible(isRecoveryVisible);
 			recoveryDate.setVisible(isRecoveryVisible);
+
```

Comment:
Same problem occurs for vaccination related field "number of doses".
Better use `FieldHelper.setVisbleWhen` with `clearOnHidden = true`.